src/train/extractor/extractor.py

```diff
@@ -69,27 +70,43 @@ def extract(self, query_results, energy_components, feature_group, energy_source
         # 1. compute energy different per timestamp and concat all energy component and unit
         power_data = self.get_power_data(query_results, energy_components, energy_source)
         if power_data is None:
-            return None, None, None
+            return None, None, None, None
         power_data = drop_zero_column(power_data, power_data.columns)
         power_columns = power_data.columns
         features = FeatureGroups[FeatureGroup[feature_group]]
         # 2. separate workload and system 
         workload_features = [feature for feature in features if feature not in SYSTEM_FEATURES]
         system_features = [feature for feature in features if feature in SYSTEM_FEATURES]
+        # 2.1 separate accelerator workload form workloads
+        accelerator_features = [feature for feature in features if feature in FeatureGroups[FeatureGroup.AcceleratorOnly]]
+        workload_features = [feature for feature in features if feature not in FeatureGroups[FeatureGroup.AcceleratorOnly]]
         # 3. compute aggregated utilization different per timestamp and concat them
-        feature_data = self.get_workload_feature_data(query_results, workload_features)
-        if feature_data is None:
-            return None, None, None
+        cur_accelerator_features = []
+        accelerator_flag  = False
+
+        if len(workload_features) == 0 and len(accelerator_features) > 0:
+            accelerator_flag = True
+            accelerator_feature_data, cur_accelerator_features = self.get_accelerator_feature_data(query_results, accelerator_features)
+            feature_data = accelerator_feature_data.reset_index()
+            cur_features = cur_accelerator_features
+        else:
+            feature_data = self.get_workload_feature_data(query_results, workload_features)
+            cur_features = workload_features
+            if feature_data is None:
+                return None, None, None,None
+
         # join power
         feature_power_data = feature_data.set_index(TIMESTAMP_COL).join(power_data).sort_index().dropna()
         
         # aggregate data if needed
         is_aggr = node_level and aggr
         if is_aggr:
             # sum stat of all containers
-            sum_feature = feature_power_data.groupby([TIMESTAMP_COL]).sum()[workload_features]
```

Comment:
Keep using the `workload_features`. FeatureGroup must be redefined for each feature column at the beginning.